roles/satellite-clone/tasks/reset_pulp_data.yml

```diff
@@ -1,17 +1,53 @@
-- name: clean pulp database
+- name: clean pulp2 database
   command: mongo pulp_database --eval "db.units_distribution.drop();db.units_docker_blob.drop();db.units_docker_image.drop();db.units_docker_manifest.drop();db.units_docker_tag.drop();db.units_drpm.drop();db.units_erratum.drop();db.units_iso.drop();db.units_package_category.drop();db.units_package_environment.drop();db.units_package_group.drop();db.units_puppet_module.drop();db.units_rpm.drop();db.units_srpm.drop();db.units_yum_repo_metadata_file.drop();db.repo_content_units.drop();"
+  when:
+    - satellite_version in ["6.6", "6.7", "6.8", "6.9"]
 
-- name: reset pulp repo importers
+- name: reset pulp2 repo importers
   command: 'mongo pulp_database --eval "db.repo_importers.update({\"scratchpad.repomd_revision\": {$exists: true}}, {$set: {\"scratchpad.repomd_revision\": null}}, {multi: true});"'
+  when:
+    - satellite_version in ["6.6", "6.7", "6.8", "6.9"]
 
-- name: reset pulp data
-  command: echo "Katello::Rpm.all.destroy_all; Katello::Erratum.all.destroy_all; Katello::PackageGroup.all.destroy_all; Katello::PuppetModule.all.destroy_all; Katello::DockerManifest.all.destroy_all; Katello::DockerTag.all.destroy_all" | foreman-rake console
+- name: reset pulp data in Katello
+  command: echo "Katello::Rpm.all.destroy_all; Katello::Erratum.all.destroy_all; Katello::PackageGroup.all.destroy_all; Katello::DockerManifest.all.destroy_all; Katello::DockerTag.all.destroy_all" | foreman-rake console
 
-- name: stop pulp services
-  command: foreman-maintain service stop --only httpd,pulp_workers,pulp_celerybeat,pulp_resource_manager,pulp_streamer
+- name: reset pulp2-specific data in Katello
+  command: echo "Katello::PuppetModule.all.destroy_all" | foreman-rake console
+  when:
+    - satellite_version in ["6.6", "6.7", "6.8", "6.9"]
 
-- name: migrate pulp db (reset pulp data)
+- name: reset pulpcore-specific data in Katello
+  command: echo "Katello::AnsibleCollection.all.destroy_all" | foreman-rake console
+  when:
+    - satellite_version in ["6.10", "7.0"]
+
+- name: reset pulpcore-specific data in Katello
+  command: echo "Katello::GenericContentUnit.all.destroy_all" | foreman-rake console
+  when:
+    - satellite_version in ["7.0"]
+
+- name: stop all services but the databases
+  command: foreman-maintain service stop --exclude postgresql,rh-mongodb34-mongod
+
+- name: migrate pulp2 db (reset pulp data)
   command: sudo -u apache pulp-manage-db
+  when:
+    - satellite_version in ["6.6", "6.7", "6.8", "6.9"]
+
+- name: flush pulpcore db (reset pulp data)
+  command: sudo -u pulp PULP_SETTINGS=/etc/pulp/settings.py pulpcore-manager flush --noinput
```

Comment:
Isn't there some built in magic in Ansible to do sudo? Or would that break things?

Reply:
It can, but I am not sure if it will pass the env var correctly (sudo drops the env, by default) and I kinda didn't want to experiment too much